src/java.base/share/classes/java/io/Serializable.java

```diff
@@ -43,20 +43,19 @@
  * serializable.  The serialization interface has no methods or fields
  * and serves only to identify the semantics of being serializable. <p>
  *
- * To allow subtypes of non-serializable classes to be serialized, the
- * subtype may assume responsibility for saving and restoring the
- * state of the supertype's public, protected, and (if accessible)
- * package fields.  The subtype may assume this responsibility only if
- * the class it extends has an accessible no-arg constructor to
- * initialize the class's state.  It is an error to declare a class
- * Serializable if this is not the case.  The error will be detected at
- * runtime. <p>
- *
- * During deserialization, the fields of non-serializable classes will
- * be initialized using the public or protected no-arg constructor of
- * the class.  A no-arg constructor must be accessible to the subclass
- * that is serializable.  The fields of serializable subclasses will
- * be restored from the stream. <p>
+ * It is possible for subtypes of non-serializable classes to be serialized
+ * and deserialized. During serialization, no data will be written for the
+ * fields of non-serializable classes. During deserialization, the fields of
```

Comment:
" fields of non-serializable classes" - this strictly means " fields of non-serializable SUPERclasses", right?